packages/php-wasm/universal/src/lib/base-php.ts

```diff
@@ -616,6 +616,53 @@ export abstract class BasePHP implements IsomorphicLocalPHP {
 		this[__private__dont__use].FS.rename(fromPath, toPath);
 	}
 
+	/** @inheritDoc */
+	@rethrowFileSystemError('Could not copy "{path}"')
+	cp(fromPath: string, toPath: string, recursive = false) {
+		const FS = this[__private__dont__use].FS;
+
+		const fromStat = FS.stat(fromPath);
+
+		// Emscripten directories will have the sticky-bit set
+		// https://linux.die.net/man/1/chmod
+		const fromIsDir = fromStat.mode & 0o40000;
+
+		let toExists: boolean;
```

Comment:
I wonder if this would be more readable, what do you think?

```suggestion
		let destinationExists: boolean;
```